Organise Predef methods into Scaladoc groups

janekdb opened the pull request:
This change uses @group\* tags to declutter Predef from a Scaladoc point of view.

It would be excellent to have feedback on any aspect of this but in particular the group descriptions which where oddly the most thought provoking part of this. It's difficult to be concise and useful!

> Not in scope for this PR but Predef is a bit odd because it contains methods that a user should know about and have documentation for, and other methods and things which live here to tie other parts of the system together. Maybe there should be a `InternalsDef` where these chaps can live out of view.

Preview of the page is shown below.
## V4

By squinting carefully its possible to see the group descriptions. Click the image to go large.

![predef-with-groups-v4](https://cloud.githubusercontent.com/assets/1123855/13830793/faa29af6-ebc6-11e5-9476-df79f67e8ef4.png)

SethTisue commented:
This looks marvelous. I (shamefacedly) admit I didn't even know Scaladoc *had* this feature.

Someone on Gitter said you can attach a "description" to each group? If that's true, that sounds like a good way to indicate that some of the groups contain methods that are not intended to be called directly from user code.

janekdb commented:
@groupdesc allows descriptions to be added. Suggestions for a sentence or two for each group welcomed.

janekdb commented:
Add some @groupdesc tags. They are merely placeholders at this point so please lean in with suggestions.

janekdb committed:
Organise Predef members into Scaladoc groups

By grouping members and providing descriptions the signal to noise
ratio in Predef is usefully enhanced.

janekdb commented:
Review by @SethTisue, @som-snytt, @felixmulder, @heathermiller,

SethTisue commented:
LGTM (and can always be tweaked further, anyway, if anyone sees any further improvements to be made)

VladUreche commented:
:+1: :100: :tada:

> This looks marvelous. I (shamefacedly) admit I didn't even know Scaladoc had this feature.

I must shamefacedly admit I never came around to documenting it :(

som-snytt commented:
I forgot to say this looks great.

janekdb commented:
Shamefacedly I have added a _groups_ TODO item to the Scaladoc documentation issue but not have not yet done it: https://github.com/scala/scala-lang/pull/394.

Although I do have my notes on @groups* which without too much of a shameface could be described as not unadjacent to more than marginally useful...

VladUreche commented:
> Although I do have my notes on `@groups*` which without too much of a shameface could be described as not unadjacent to more than marginally useful.

Well, that's a good start nevertheless!

> I also wonder if we could take a systemic view to applying some carefully selected groups to collections. Transformers, Views, Traversers, Mutators??? I seem to remember seeing something like this somewhere else but don't recall where now.

Yes, in theory (modulo possible bugs) the groups attribute is inherited, so we only need to add it in one place :)
